src/Swift.Runtime/src/Swift/SwiftArray.cs

```diff
@@ -21,15 +22,17 @@ public interface ISwiftCollection { }
 /// Represents a Swift array.
 /// </summary>
 /// <typeparam name="Element">The element type contained in the array.</typeparam>
-public class SwiftArray<Element> : IDisposable, ISwiftObject
+public class SwiftArray<Element> : ISwiftObject
 {
     static nuint _payloadSize = SwiftObjectHelper<SwiftArray<Element>>.GetTypeMetadata().Size;
 
     static nuint _elementSize = ElementTypeMetadata.Size;
 
-    private SwiftHandle _buffer;
+    private SwiftSafeHandle<SwiftArray<Element>> _payload;
 
-    private bool _disposed = false;
+    public SwiftSafeHandle<SwiftArray<Element>> Payload => _payload;
+
+    public unsafe IntPtr PayloadBuffer => *(IntPtr*)_payload.DangerousGetHandle();
```

Comment:
Would it be better to return an `IDisposable` that handles the AddRef/Release semantics here?

Reply:
It looks like this property is expected to be called with the AddRef/Release semantics. I'd suggest a more robuse API such as the following:

```csharp
public IDisposable GetPayloadBuffer(out IntPtr payloadBuffer);
```

The returned `IDisposable` instance would take the AddRef in the constructor and call the Release in the `Dispose()` method.

Reply:
Great idea, changed to use `GetPayloadBuffer`. Here is an example:
```csharp
    /// <summary>
    /// Gets the number of elements in the array.
    /// </summary>
    public int Count
    {
        get
        {
            using IDisposable _ = GetPayloadBuffer(out IntPtr payloadBuffer);
            int result = (int)SwiftArrayPInvokes.Count(payloadBuffer, ElementTypeMetadata);
            return result;
        }
    }
```

Reply:
The Buffer type depends on the payload type, so I created a generic `PayloadBuffer<T>` to return the correct buffer type:
```csharp
public unsafe ref struct PayloadBuffer<T> : IDisposable where T : unmanaged
{
    public T Buffer => *(T*)_payload.DangerousGetHandle();
}
```


Here's an example using `Swift.String`:
```csharp
public unsafe PayloadBuffer<SwiftString.Buffer> PayloadBuffer => new PayloadBuffer<SwiftString.Buffer>(_payload);

```
Then:
```csharp
public int Length
{
    get
    {
        using PayloadBuffer<SwiftString.Buffer> disposable = PayloadBuffer;
        return (int)PInvoke_GetLength(disposable.Buffer);
    }
}
```

Reply:
Looks great! Thank you.